plan/explain_test.go

```diff
@@ -237,12 +237,12 @@ func (s *testExplainSuite) TestExplain(c *C) {
 		{
 			"select sum(t1.c1 in (select c1 from t2)) from t1",
 			[]string{
-				"TableScan_18   cop table:t1, range:(-inf,+inf), keep order:true 8000",
-				"TableReader_19 HashSemiJoin_16  root data:TableScan_18 8000",
-				"TableScan_14   cop table:t2, range:(-inf,+inf), keep order:false 8000",
-				"TableReader_15 HashSemiJoin_16  root data:TableScan_14 8000",
-				"HashSemiJoin_16 StreamAgg_9 TableReader_19,TableReader_15 root right:TableReader_15, aux, equal:[eq(test.t1.c1, test.t2.c1)] 8000",
-				"StreamAgg_9  HashSemiJoin_16 root type:stream, funcs:sum(5_aux_0) 1",
+				"TableScan_13   cop table:t1, range:(-inf,+inf), keep order:true 8000",
+				"TableReader_14 MergeJoin_27  root data:TableScan_13 8000",
+				"IndexScan_21   cop table:t2, index:c1, range:[<nil>,+inf], out of order:false 8000",
+				"IndexReader_22 MergeJoin_27  root index:IndexScan_21 8000",
+				"MergeJoin_27 StreamAgg_9 TableReader_14,IndexReader_22 root left outer semi join, equal:[eq(test.t1.c1, test.t2.c1)], asc, left key:test.t1.c1, right key:test.t2.c1 8000",
```

Comment:
Distinguish `MergeSemiJoin` from `MergeJoin`

Reply:
I distinguished them by operator name in **plan/stringer.go**. In explain info we already have join type to distinguish them, so I keep the operator name to "MergeJoin"